index.js

```diff
@@ -5,7 +5,16 @@ var isglob = require('is-glob');
 var pathDirname = require('path-dirname');
 
 module.exports = function globParent(str) {
-	str += 'a'; // preserves full path in case of trailing path separator
+	// replace braces/brackets sections with *
+	str = str.replace(/(^|[^\\])(\{([^{}]*?)}|\[([^\[\]]*?)\])/g, '$1*');
+	if (/[\}\}]$/.test(str)) str += '/';
```

Comment:
This only requires the first one to be escaped, but yeah I didn't look at any nested cases.

Think I should abandon this approach, or can it possibly be tweaked to do the job?

Either way seems I should add some nested braces cases to the tests.